test/functional/TestUtilities/src/org/openj9/test/util/StringUtilities.java

```diff
@@ -38,10 +39,19 @@
 	 * @return Optional which is either empty or contains matching string 
 	 */
 	public static Optional<String> searchSubstring(String needle, List<String> haystack) {
-		return haystack.stream().filter(s -> s.contains(needle)).findFirst();
+		return searchSubstring(needle, haystack.stream());
+	}
+	
+	/**
+	 * Find the first occurrence of a substring in a list of Strings.
+	 * @param needle pattern for which to search
+	 * @param haystack list of Strings to search
+	 * @return Optional which is either empty or contains matching string 
+	 */
+	public static Optional<String> searchSubstring(String needle, Stream<String> haystack) {
+		return haystack.filter(s -> s.contains(needle)).findFirst();
```

Comment:
Since all references to this method just need `searchResult.isPresent()`, this method can return a boolean result instead.

Reply:
This is intended as a general utility.  Callers may want the full String containing the needle.